core-plugins/src/main/java/co/cask/hydrator/plugin/batch/aggregator/function/CollectSet.java

```diff
@@ -0,0 +1,47 @@
+package co.cask.hydrator.plugin.batch.aggregator.function;
+
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Collect Set of a specific column
+ * @param <T> type of aggregate value
+ */
+public class CollectSet<T> implements AggregateFunction<Set<T>> {
+
+    private Set<T> collectSet;
+    private final String fieldName;
+    private final Schema fieldSchema;
+
+    public CollectSet(String fieldName, Schema fieldSchema) {
+        this.fieldName = fieldName;
+        this.fieldSchema = fieldSchema;
+    }
+
+    @Override
+    public void beginFunction() {
+        collectSet = new HashSet<>();
+    }
+
+    @Override
+    public void operateOn(StructuredRecord record) {
+        T field = record.get(fieldName);
+        if (Objects.nonNull(field)) {
```

Comment:
how come the set filters out nulls but the list does not? Seems like they should be consistent. My initial thought is that they should both include nulls.

Reply:
Sure, i'll remove null checking in Set as well

Reply:
Fiixed